composer/callbacks/early_stopper.py

```diff
@@ -0,0 +1,139 @@
+# Copyright 2021 MosaicML. All Rights Reserved.
+
+"""Early stopping callback."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Callable, Optional, Union
+
+import numpy as np
+
+from composer.core import State, Time
+from composer.core.callback import Callback
+from composer.core.time import TimeUnit
+from composer.loggers import Logger
+
+log = logging.getLogger(__name__)
+
+__all__ = ["EarlyStopper"]
+
+
+class EarlyStopper(Callback):
+    """This callback tracks a training or evaluation metric and halts training if the metric does not
+    improve within a given interval.
+
+    Example
+
+    .. doctest::
+
+        >>> from composer.callbacks.early_stopper import EarlyStopper
+        >>> from torchmetrics.classification.accuracy import Accuracy
+        >>> # constructing trainer object with this callback
+        >>> early_stopper = EarlyStopper("Accuracy", "my_evaluator", patience=1)
+        >>> evaluator = Evaluator(
+        ...     dataloader = eval_dataloader,
+        ...     label = 'my_evaluator',
+        ...     metrics = Accuracy()
+        ... )
+        >>> trainer = Trainer(
+        ...     model=model,
+        ...     train_dataloader=train_dataloader,
+        ...     eval_dataloader=evaluator,
+        ...     optimizers=optimizer,
+        ...     max_duration="1ep",
+        ...     callbacks=[early_stopper],
+        ... )
+
+    Args:
+        monitor (str): The name of the metric to monitor.
+        dataloader_label (str): The label of the dataloader or evaluator associated with the tracked metric. If 
+            monitor is in an Evaluator, the dataloader_label field should be set to the Evaluator's label. If 
+            monitor is a training metric or an ordinary evaluation metric not in an Evaluator, dataloader_label
+            should be set to 'train' or 'eval' respectively.
+        comp (Callable[[Any, Any], bool], optional): A comparison operator to measure change of the monitored metric. The comparison
```

Comment:
Is it possible to use torch.less / torch.greater be used instead of numpy (in case if metric tensors are on GPU?)